packages/@aws-cdk/aws-lambda/lib/event-source-mapping.ts

```diff
@@ -120,6 +120,13 @@ export interface EventSourceMappingOptions {
    */
   readonly startingPosition?: StartingPosition;
 
+  /**
+   * A list of current response types applied to the event source mapping.
+   *
+   * @default - None
+   */
+  readonly functionResponseTypes?: FunctionResponseType[];
```

Comment:
It would be a more natural experience if we switched to a struct and flattened the nesting.

Currently, the user would do -

```ts
functionResponseTypes: [FunctionResponseType.REPORT_BATCH_ITEM_FAILURES]
```

Instead, we should change this to -

```ts
reportBatchItemFailures: true,
```

If I understand correctly, this is similar to the AWS console experience.

Reply:
Sorry for the late reply. I missed the notification apparently. I like the suggestion you made and incorporated this into the PR.